2 tests fail in cron travis run

e.g. from https://app.travis-ci.com/github/datalad/datalad/jobs/587920121

```
=========================== short test summary info ============================
FAILED ../datalad/customremotes/tests/test_datalad.py::test_basic_scenario_local_url - AssertionError: assert '_' in '3versions-allversioned.txt'
FAILED ../datalad/distributed/tests/test_ria_basics.py::test_version_check - datalad.support.exceptions.IncompleteResultsError: Command did not complete successfully. 1 failed:
[{'action': 'copy',
  'annexkey': 'MD5E-s18--6beb3be625001976a17c6eb78f88c592',
  'error_message': 'Remote is treated as read-only. Set '
                   "'ora-remote.<name>.force-write=true' to overrule this.\n"
                   'This could have failed because --fast is enabled.',
  'path': '/tmp/datalad_temp__test_version_check8aj6f1ae/new_file',
  'refds': '/tmp/datalad_temp__test_version_check8aj6f1ae',
  'status': 'error',
  'target': 'store',
  'type': 'file'}]
= 2 failed, 1160 passed, 45 skipped, 4 xfailed, 17 warnings in 869.72s (0:14:29) =
The command "http_proxy= PATH=$PWD/../tools/coverage-bin:$PATH $PYTEST_WRAPPER python -m pytest "${PYTEST_OPTS[@]}" -c ../tox.ini -n 2 -m "${PYTEST_SELECTION:+$PYTEST_SELECTION_OP($PYTEST_SELECTION) and }not(turtle)" --doctest-modules --cov=datalad --cov-report= --pyargs $TESTS_TO_PERFORM" exited with 1.
```

<details>
<summary>had been happening for awhile, e.g. all previous month</summary> 

```shell
(git)smaug:/mnt/datasets/datalad/ci/logs/2022/10[master]git
$> git grep 'FAILED.*test_basic_scenario_local_url'
07/cron/20221007T213523/22c97cd/travis-14676-failed/12.txt:[gw1] [ 17%] FAILED ../datalad/customremotes/tests/test_datalad.py::test_basic_scenario_local_url 
07/cron/20221007T213523/22c97cd/travis-14676-failed/12.txt:FAILED ../datalad/customremotes/tests/test_datalad.py::test_basic_scenario_local_url
11/cron/20221011T215806/ddd9b7b/travis-14707-failed/12.txt:[gw1] [ 17%] FAILED ../datalad/customremotes/tests/test_datalad.py::test_basic_scenario_local_url 
11/cron/20221011T215806/ddd9b7b/travis-14707-failed/12.txt:FAILED ../datalad/customremotes/tests/test_datalad.py::test_basic_scenario_local_url
14/cron/20221014T194516/9d4ba24/travis-14733-failed/12.txt:[gw1] [ 17%] FAILED ../datalad/customremotes/tests/test_datalad.py::test_basic_scenario_local_url 
14/cron/20221014T194516/9d4ba24/travis-14733-failed/12.txt:FAILED ../datalad/customremotes/tests/test_datalad.py::test_basic_scenario_local_url
16/pr/7085/e2ffa2e/travis-14734-failed/3.txt:[gw1] [  3%] FAILED ../customremotes/tests/test_datalad.py::test_basic_scenario_local_url 
16/pr/7085/e2ffa2e/travis-14734-failed/3.txt:FAILED ../customremotes/tests/test_datalad.py::test_basic_scenario_local_url
16/pr/7085/e2ffa2e/travis-14734-failed/4.txt:[gw1] [  3%] FAILED ../customremotes/tests/test_datalad.py::test_basic_scenario_local_url 
16/pr/7085/e2ffa2e/travis-14734-failed/4.txt:FAILED ../customremotes/tests/test_datalad.py::test_basic_scenario_local_url
18/cron/20221018T215935/54e21e3/travis-14760-failed/12.txt:[gw1] [ 17%] FAILED ../datalad/customremotes/tests/test_datalad.py::test_basic_scenario_local_url 
18/cron/20221018T215935/54e21e3/travis-14760-failed/12.txt:FAILED ../datalad/customremotes/tests/test_datalad.py::test_basic_scenario_local_url
21/cron/20221021T194615/2346326/travis-14776-failed/12.txt:[gw1] [ 17%] FAILED ../datalad/customremotes/tests/test_datalad.py::test_basic_scenario_local_url 
21/cron/20221021T194615/2346326/travis-14776-failed/12.txt:FAILED ../datalad/customremotes/tests/test_datalad.py::test_basic_scenario_local_url
25/cron/20221025T220312/dfc97e7/travis-14809-failed/12.txt:[gw1] [ 13%] FAILED ../datalad/customremotes/tests/test_datalad.py::test_basic_scenario_local_url 
25/cron/20221025T220312/dfc97e7/travis-14809-failed/12.txt:FAILED ../datalad/customremotes/tests/test_datalad.py::test_basic_scenario_local_url - AssertionError: assert '_' in '3versions-allversioned.txt'
28/cron/20221028T194650/2346326/travis-14824-failed/12.txt:[gw1] [ 13%] FAILED ../datalad/customremotes/tests/test_datalad.py::test_basic_scenario_local_url 
28/cron/20221028T194650/2346326/travis-14824-failed/12.txt:FAILED ../datalad/customremotes/tests/test_datalad.py::test_basic_scenario_local_url - AssertionError: assert '_' in '3versions-allversioned.txt'
git grep 'FAILED.*test_basic_scenario_local_url'  4.76s user 4.14s system 330% cpu 2.691 total
1 9758.....................................:Tue 08 Nov 2022 10:07:35 PM EST:.
(git)smaug:/mnt/datasets/datalad/ci/logs/2022/10[master]git
$> git grep 'FAILED.*test_ria_basics.py::test_version_check'                              
07/cron/20221007T213523/22c97cd/travis-14676-failed/12.txt:[gw1] [ 22%] FAILED ../datalad/distributed/tests/test_ria_basics.py::test_version_check 
07/cron/20221007T213523/22c97cd/travis-14676-failed/12.txt:FAILED ../datalad/distributed/tests/test_ria_basics.py::test_version_check - ...
11/cron/20221011T215806/ddd9b7b/travis-14707-failed/12.txt:[gw1] [ 22%] FAILED ../datalad/distributed/tests/test_ria_basics.py::test_version_check 
11/cron/20221011T215806/ddd9b7b/travis-14707-failed/12.txt:FAILED ../datalad/distributed/tests/test_ria_basics.py::test_version_check - ...
14/cron/20221014T194516/9d4ba24/travis-14733-failed/12.txt:[gw1] [ 22%] FAILED ../datalad/distributed/tests/test_ria_basics.py::test_version_check 
14/cron/20221014T194516/9d4ba24/travis-14733-failed/12.txt:FAILED ../datalad/distributed/tests/test_ria_basics.py::test_version_check - ...
18/cron/20221018T215935/54e21e3/travis-14760-failed/12.txt:[gw1] [ 22%] FAILED ../datalad/distributed/tests/test_ria_basics.py::test_version_check 
18/cron/20221018T215935/54e21e3/travis-14760-failed/12.txt:FAILED ../datalad/distributed/tests/test_ria_basics.py::test_version_check - ...
21/cron/20221021T194615/2346326/travis-14776-failed/12.txt:[gw1] [ 22%] FAILED ../datalad/distributed/tests/test_ria_basics.py::test_version_check 
21/cron/20221021T194615/2346326/travis-14776-failed/12.txt:FAILED ../datalad/distributed/tests/test_ria_basics.py::test_version_check - ...
25/cron/20221025T220312/dfc97e7/travis-14809-failed/12.txt:[gw1] [ 24%] FAILED ../datalad/distributed/tests/test_ria_basics.py::test_version_check 
28/cron/20221028T194650/2346326/travis-14824-failed/12.txt:[gw1] [ 24%] FAILED ../datalad/distributed/tests/test_ria_basics.py::test_version_check 
git grep 'FAILED.*test_ria_basics.py::test_version_check'  3.89s user 2.62s system 857% cpu 0.759 total

```
</details>

it is the cron run with `_DL_ANNEX_INSTALL_SCENARIO="miniconda --python-match minor --batch git-annex=8.20200309 -m conda"`

--- a/changelog.d/pr-7159.md
+++ b/changelog.d/pr-7159.md
@@ -1,0 +1,3 @@
+### 🧪 Tests
+
+- Fix test failure with old annex.  Fixes [#7157](https://github.com/datalad/datalad/issues/7157) via [PR #7159](https://github.com/datalad/datalad/pull/7159) (by [@bpoldrack](https://github.com/bpoldrack))

--- a/datalad/distributed/ora_remote.py
+++ b/datalad/distributed/ora_remote.py
@@ -1117,7 +1117,7 @@ class RIARemote(SpecialRemote):
         self.remote_git_dir, self.remote_archive_dir, self.remote_obj_dir = \
             self.get_layout_locations(store_base_path, self.archive_id)
 
-        read_only_msg = "Treating remote as read-only in order to" \
+        read_only_msg = "Treating remote as read-only in order to " \
                         "prevent damage by putting things into an unknown " \
                         "version of the target layout. You can overrule this " \
                         "by setting 'annex.ora-remote.<name>.force-write=true'."

--- a/datalad/distributed/tests/test_ria_basics.py
+++ b/datalad/distributed/tests/test_ria_basics.py
@@ -31,7 +31,10 @@ from datalad.distributed.tests.ria_utils import (
     get_all_files,
     populate_dataset,
 )
-from datalad.support.exceptions import CommandError
+from datalad.support.exceptions import (
+    CommandError,
+    IncompleteResultsError,
+)
 from datalad.tests.utils_pytest import (
     SkipTest,
     assert_equal,
@@ -470,9 +473,7 @@ def _test_version_check(host, dspath, store):
         f.write("arbitrary addition")
     ds.save(message="Add a new_file")
 
-    # TODO: use self.annex.error in special remote and see whether we get an
-    #       actual error result
-    with assert_raises(CommandError):
+    with assert_raises((CommandError, IncompleteResultsError)):
         ds.push('new_file', to='store')
 
     # However, we can force it by configuration
@@ -486,6 +487,7 @@ def _test_version_check(host, dspath, store):
 def test_version_check_ssh():
     # TODO: Skipped due to gh-4436
     _test_version_check('datalad-test')
+
 
 def test_version_check():
     _test_version_check(None)

